src/sprite_resource.c

```diff
@@ -0,0 +1,884 @@
+#include "sprite_resource.h"
+
+#include <nitro.h>
+#include <string.h>
+
+#include "constants/heap.h"
+
+#include "struct_decls/struct_02006C24_decl.h"
+
+#include "nnsys/g2d/g2d_Image.h"
+
+#include "heap.h"
+#include "narc.h"
+#include "resource_collection.h"
+#include "unk_02006E3C.h"
+
+#define SPRITE_VRAM_TYPE_DEFAULT NNS_G2D_VRAM_TYPE_3DMAIN
+#define NARC_INDEX_NONE          0xFFFFFFFE
+
+typedef struct TileResourceData {
+    NNSG2dCharacterData *tileData;
+    NNS_G2D_VRAM_TYPE vramType;
+} TileResourceData;
+
+typedef struct PaletteResourceData {
+    NNSG2dPaletteData *paletteData;
+    NNS_G2D_VRAM_TYPE vramType;
+    int paletteIndex;
+} PaletteResourceData;
+
+typedef struct SpriteResourceData {
+    NNSG2dCellDataBank *spriteBank;
+} SpriteResourceData;
+
+typedef struct SpriteAnimResourceData {
+    NNSG2dAnimBankData *animBank;
+} SpriteAnimResourceData;
+
+typedef struct MultiSpriteResourceData {
+    NNSG2dMultiCellDataBank *multiSpriteBank;
+} MultiSpriteResourceData;
+
+typedef struct MultiSpriteAnimResourceData {
+    NNSG2dAnimBankData *multiSpriteAnimBank;
+} MultiSpriteAnimResourceData;
+
+typedef struct SpriteResourceTableBinary {
+    enum SpriteResourceType type;
+    union {
+        SpriteResourceTableEntryFile fileEntries[1];
+        SpriteResourceTableEntryNARC narcEntries[1];
```

Comment:
**question:** Is the subscript necessary to match?

Reply:
No, but in my opinion it makes the code that actually *uses* these fields nicer. Could probably do `[0]` as well, provided mwcc supports that.

Reply:
```suggestion
        SpriteResourceTableEntryFile *fileEntries;
        SpriteResourceTableEntryNARC *narcEntries;
```

Reply:
That's not the same code tho. It's an inline array, not a pointer to an array.